dev_guide/managing_images.adoc

```diff
@@ -0,0 +1,690 @@
++
+----
+$ oc whoami -t
+----
+
+. Log in to the internal registry using the token. You must have *docker*
+installed on your system:
++
+----
+$ docker login -u <user_name> -e <email_address> \
+    -p <token_value> <registry_location>:<port>
+----
++
+[NOTE]
+====
+Contact your cluster administrator if you do not know the registry IP or host
+name and port to use.
+====
+
+In order to pull an image, the authenticated user must have `get` rights on the
+requested `imagestreams/layers`. In order to push an image, the authenticated
+user must have `update` rights on the requested `imagestreams/layers`.
+
+By default, all service accounts in a project have rights to pull any image in
+the same project, and the *builder* service account has rights to push any image
+in the same project.
+
+[[using-image-pull-secrets]]
+== Using Image Pull Secrets
+
+link:../architecture/infrastructure_components/image_registry.html[Docker
+registries] can be secured to prevent unauthorized parties from accessing
+certain images. If you are link:#accessing-the-internal-registry[using
+{product-title}'s internal registry] and are pulling from image streams located
+in the same project, then your pod's service account should already have
+permissions and no additional action should be required. If this is not the
+case, then additional configuration steps are required.
+
+[[allowing-pods-to-reference-images-across-projects]]
+=== Allowing Pods to Reference Images Across Projects
+
+When using the internal registry, to allow pods in *project-a* to reference
+images in *project-b*, a service account in *project-a* must be bound to the
+`system:image-puller` role in *project-b*:
+
+----
+$ oc policy add-role-to-user \
+    system:image-puller system:serviceaccount:project-a:default \
+    --namespace=project-b
+----
+
+After adding that role, the pods in *project-a* that reference the default
+service account will be able to pull images from *project-b*.
+
+To allow access for any service account in *project-a*, use the group:
+
+----
+$ oc policy add-role-to-group \
+    system:image-puller system:serviceaccounts:project-a \
+    --namespace=project-b
+----
+
+[[allowing-pods-to-reference-images-from-other-secured-registries]]
+=== Allowing Pods to Reference Images from Other Secured Registries
+
+To pull a secured Docker image that is not from OpenShift's internal registry,
+you must create a *dockercfg* secret and add it to your service account.
+
+If you already have a *_.dockercfg_* file for the secured registry, you can
+create a secret from that file by running:
+
+----
+$ oc secrets new <pull_secret_name> .dockercfg=<path/to/.dockercfg>
+----
+
+If you do not already have a *_.dockercfg_* file for the secured registry, you
+can create a secret by running:
+
+----
+$ oc secrets new-dockercfg <pull_secret_name> \
+    --docker-server=<registry_server> --docker-username=<user_name> \
+    --docker-password=<password> --docker-email=<email>
+----
+
```

Comment:
Mostly newer docker client create `$HOME/.docker/config.json` rather than `$HOME/.dockercfg` for that it would be nice to add this command `oc secrets new SECRET .dockerconfigjson=path/to/.docker/config.json` which does from that new file.